Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/GDSError/GDSErrorViewController.swift

```diff
@@ -0,0 +1,89 @@
+///  - A `viewWillAppear` lifecycle event triggers the `didAppear` method in the viewModel.
+///  - A `dismissScreen` lifecycle event triggers the `didDismiss` method in the viewModel.
+///
+/// The `primaryButton`'s action is set from the ``ButtonViewModel`` in the viewModel.
+/// The `secondaryButton`'s action is set from the ``ButtonViewModel`` in the viewModel.
+public final class GDSErrorViewController: BaseViewController {
+    public override var nibName: String? { "GDSError" }
+    
+    private let viewModel: GDSErrorViewModel
+
+    public init(viewModel: GDSErrorViewModel) {
+        self.viewModel = viewModel
+        super.init(viewModel: viewModel as? BaseViewModel, nibName: "Error", bundle: .module)
+    }
+    
+    @available(*, unavailable, renamed: "init(coordinator:)")
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    @IBOutlet private var errorImageView: UIImageView! {
+        didSet {
+            let font = UIFont(style: .largeTitle, weight: .light)
+            let configuration = UIImage.SymbolConfiguration(font: font, scale: .large)
+            errorImageView.image = UIImage(systemName: viewModel.image, withConfiguration: configuration)
+            errorImageView.accessibilityIdentifier = "error-image"
+        }
+    }
+    
+    @IBOutlet private var errorTitle: UILabel! {
+        didSet {
+            errorTitle.font = .init(style: .largeTitle, weight: .bold, design: .default)
+            errorTitle.text = viewModel.title.value
+            errorTitle.accessibilityIdentifier = "error-title"
+        }
+    }
+    
+    @IBOutlet private var errorBody: UILabel! {
+        didSet {
+            errorBody.text = viewModel.body.value
+            errorBody.accessibilityIdentifier = "error-body"
+        }
+    }
+
+    @IBOutlet private var primaryButton: RoundedButton! {
+        didSet {
+            primaryButton.setTitle(viewModel.primaryButtonViewModel.title, for: .normal)
+            primaryButton.accessibilityIdentifier = "error-primary-button"
+        }
+    }
+    
+    @IBAction private func primaryButtonAction(_ sender: Any) {
+        primaryButton.isLoading = true
+        viewModel.primaryButtonViewModel.action()
+        primaryButton.isLoading = false
+    }
+    
+    @IBOutlet private var secondaryButton: SecondaryButton! {
+        didSet {
+            if let buttonViewModel = viewModel.secondaryButtonViewModel {
+                secondaryButton.setTitle(buttonViewModel.title, for: .normal)
+                secondaryButton.accessibilityIdentifier = "error-secondary-button"
+                if let icon = buttonViewModel.icon {
+                    secondaryButton.symbolPosition = icon.symbolPosition
+                    secondaryButton.icon = icon.iconName
+                }
+            } else {
+                secondaryButton.isHidden = true
+            }
```

Comment:
Using an `if let` here can work, but really there should be an `.isHidden = false` in the main block.

However, the `if let` is not required. Instead you could do:

```
secondaryButton.setTitle(buttonViewModel.title, for: .normal)
secondaryButton.accessibilityIdentifier = "error-secondary-button"
if let icon = buttonViewModel.icon {
  secondaryButton.symbolPosition = icon.symbolPosition
  secondaryButton.icon = icon.iconName
}
secondaryButton.isHidden = viewModel.secondaryButtonViewModel == nil
```

To break that down, the `viewModel.secondaryButtonViewModel == nil` is a boolean check that will return `true` if the button view model is `nil` and `false` if it is not `nil`.

Therefore, if the button view model is `nil` then the boolean returns `true` which sets the `.isHidden` to `true`. And if it returns `false`, then `isHidden` is set to `false` eg, it is visible :)

Reply:
I see the problem. I think we should change the `GDSLocalisedString` version of the `setTitle` method to take an optional like it does for a normal string:

<img width="604" alt="image" src="https://github.com/govuk-one-login/mobile-ios-common/assets/104073334/b8872467-2686-4a33-ad9f-48767f129686">

Then we can do:
`secondaryButton.setTitle(viewModel.secondaryButtonViewModel.title, for: .normal)` 
without the if let.